scripts/run_frontend_tests.py

```diff
@@ -89,14 +89,24 @@ def main(args=None):
             'start', os.path.join('core', 'tests', 'karma.conf.ts')]
 
     task = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-    out, _ = task.communicate()
-    task.wait()
+    output_lines = []
+    # Reads and prints realtime output from the subprocess until it terminates.
+    while True:
+        line = task.stdout.readline()
+
+        # No more output from the subprocess, and the subprocess has ended.
+        if len(line) == 0 and task.poll() is not None:
+            break
+
+        if line:
+            python_utils.PRINT(line, end=u'')
+            output_lines.append(line)
+    concatenated_output = ''.join(
+        python_utils.UNICODE(line, 'utf-8') for line in output_lines)
```

Comment:
One thing we generally do for this sort of thing is something like line.encode('utf-8') or similar, which seems to be safe and fairly foolproof. Could you please test approaches like that rather than doing a unicode() cast? I.e. figure out explicitly what type of string it is and then deal with it correctly.

If you're able to handle this in the other scripts too (like linting), that would be great -- I think this might have been due to the move to GitHub Actions which might use different string defaults. /cc @kevintab95 

Thanks!

Reply:
Done, used `line.decode('utf-8')` instead.